pkg/deploy/deploy.go

```diff
@@ -402,6 +412,44 @@ func ApplyParams(componentPath string, imageParamsMap map[string]string, isUpdat
 	return err
 }
 
+// removeResourcesFromDeployment checks if the provided resource is a Deployment,
+// and if so, removes the resources field from each container in the Deployment. This ensures we do not overwrite the
+// resources field when Patch is applied with the returned unstructured resource.
+func removeResourcesFromDeployment(u *unstructured.Unstructured) (*unstructured.Unstructured, error) {
```

Comment:
No need to return the object, you change it inplace.

Reply:
+1